### DISCUSSION TITLE
Raw mol block from SDF parsing

### DISCUSSION THREAD

**Discussion by roivant-matts** [rdkit/rdkit#6537] [Q&A] [answered]
I have a use case where I am using rdkit more for data engineering than chemistry functions. I am loading SDF files to a database where I would like to preserve the exact mol blocks and properties that are coming from an SDF.

* using SDMolSupplier to iterate though mols
* grabbing properties as dictionary
* getting mol block from MolToV3KMolBlock

The MolToV3KMolBlock will make some changes to certain molecules vs the input mol block in the SDF file (e.g. some cases where parity changes vs input). My goal is to grab the mol block from the SDF input without any processing. I could look for alternatives or write my own parser, but wanted to check if it is possible with rdkit before proceeding.

**Comment by bp-kelley** [comment, marked as answer]
SDMolSupplier has random access, and you can get the raw data as follows (con: the mols have to be written to a file):

```
with Chem.SDMolSupplier("foo.sdf") as sdf:
    for i, mol in enumerate(sdf):
        print(sdf.GetItemText(i))
```

**Comment by bp-kelley** [comment]
Note you don't have to actually parse the molecules:

```
with Chem.SDMolSupplier("foo.sdf") as sdf:
    i = 0
    while 1:
        try:
            print(sdf.GetItemText(i))
            i += 1
        except RuntimeError as e:
            if "Index error" in str(e):
                break
            raise     
```

**Comment by roivant-matts** [comment]
Exactly what I needed - appreciate it.
